src/Raven.Server/Documents/Handlers/StreamingHandler.cs

```diff
@@ -86,13 +88,22 @@ public Task SteamQueryHead()
                     catch (IndexDoesNotExistException)
                     {
                         HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
-                        context.Write(writer, json: new DynamicJsonValue
+                        if (writer.SupportError)
```

Comment:
What does this mean? We should always report an error

Reply:
how should we write a CSV error? our response body isn't a JSON

Reply:
You output the error as is, which will corrupt the CSV file, but it will give us an error that they can see

Reply:
note that QueryResultServerSide<T>  has the same notion

Reply:
there is just throws if it isn't set, which expose the error